libcudacxx/include/cuda/std/detail/libcxx/include/__utility/pair.h

```diff
@@ -69,707 +70,704 @@
+                                           && _LIBCUDACXX_TRAIT(is_nothrow_move_assignable, _T2))
+  {
+    first  = _CUDA_VSTD::forward<_T1>(__p.first);
+    second = _CUDA_VSTD::forward<_T2>(__p.second);
+    return *this;
+  }
+
+  template <class _U1, class _U2>
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr __pair_base(_U1&& __t1, _U2&& __t2) noexcept(
+    _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T1, _U1) && _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T2, _U2))
+      : first(_CUDA_VSTD::forward<_U1>(__t1))
+      , second(_CUDA_VSTD::forward<_U2>(__t2))
+  {}
+
+protected:
+  template <class... _Args1, class... _Args2, size_t... _I1, size_t... _I2>
+  _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_CONSTEXPR_AFTER_CXX17 __pair_base(
+    piecewise_construct_t,
+    tuple<_Args1...>& __first_args,
+    tuple<_Args2...>& __second_args,
+    __tuple_indices<_I1...>,
+    __tuple_indices<_I2...>);
+};
 
+template <class _T1, class _T2>
+struct _LIBCUDACXX_TEMPLATE_VIS pair : public __pair_base<_T1, _T2>
+{
+  using __base = __pair_base<_T1, _T2>;
+
+  typedef _T1 first_type;
+  typedef _T2 second_type;
+
+  template <class _Constraints                                                 = __pair_constraints<_T1, _T2>,
+            __enable_if_t<_Constraints::__implicit_default_constructible, int> = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr pair() noexcept(_LIBCUDACXX_TRAIT(is_nothrow_default_constructible, _T1)
+                                                          && _LIBCUDACXX_TRAIT(is_nothrow_default_constructible, _T2))
+      : __base()
+  {}
+
+  template <class _Constraints                                                 = __pair_constraints<_T1, _T2>,
+            __enable_if_t<_Constraints::__explicit_default_constructible, int> = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY explicit constexpr pair() noexcept(
+    _LIBCUDACXX_TRAIT(is_nothrow_default_constructible, _T1)
+    && _LIBCUDACXX_TRAIT(is_nothrow_default_constructible, _T2))
+      : __base()
+  {}
+
+  // element wise constructors
+  template <class _Constraints                                                     = __pair_constraints<_T1, _T2>,
+            __enable_if_t<_Constraints::__explicit_copy_constructible_from_T, int> = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY explicit constexpr pair(const _T1& __t1, const _T2& __t2) noexcept(
+    _LIBCUDACXX_TRAIT(is_nothrow_copy_constructible, _T1) && _LIBCUDACXX_TRAIT(is_nothrow_copy_constructible, _T2))
+      : __base(__t1, __t2)
+  {}
+
+  template <class _Constraints                                                     = __pair_constraints<_T1, _T2>,
+            __enable_if_t<_Constraints::__implicit_copy_constructible_from_T, int> = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr pair(const _T1& __t1, const _T2& __t2) noexcept(
+    _LIBCUDACXX_TRAIT(is_nothrow_copy_constructible, _T1) && _LIBCUDACXX_TRAIT(is_nothrow_copy_constructible, _T2))
+      : __base(__t1, __t2)
+  {}
```

Comment:
In other places the implicit version is at the top and the explicit one is at the bottom of the pair; can you harmonize this throughout this file?